.github/composite_actions/log_cw_metric_wrapper/action.yaml

```diff
@@ -85,8 +85,10 @@ runs:
 
         metric-name: "github_metric_1.0"
         test-type: ${{ inputs.test-type }}
-        package-name: ${{ inputs.package-name }}
-        matrix: ${{ toJson(matrix) }}
+        working-directory: ${{ inputs.working-directory }}
+
+        # Sending a null value will cause the action to fail so we must send an empty string here.
+        matrix: ${{ matrix == '' && '{}' || toJSON(matrix) }}
```

Comment:
Question asked out of ignorance: Where does the `matrix` var come from, and why wouldn't we do ${{ toJSON(matrix) || matrix == '' && '{}'  }}?

Reply:
`matrix` is a github context variable see: https://docs.github.com/en/actions/learn-github-actions/contexts#matrix-context

I guess we could do it that way as it's just an inversion of the logic.

Reply:
I read this statement as:

> if `matrix` equals "{}", assign the matrix yaml value to "true".
> ELSE
> parse the matrix yaml value from json

I don't understand that logic. Why would we set the matrix yaml value to "true"? Or am I misreading this? I'm very new to GH actions so it's entirely possible I am.

Reply:
We don't have ternary operators in Github Actions but that's essentially what we're trying to do here. If matrix is null we want to send {} otherwise we send a json version of matrix.  

To do that we first check if matrix environment variable is empty or not.  If it's empty we send {} this is achieved via &&, otherwise (||) we send toJSON(matrix). 

Yes it is a javascript style logical operators.   It is not evaluating to both '' and '{}' at the same time, it is using the '' to check matrix value before returning {} in that case. 

`${{ matrix == '' && '{}' || toJSON(matrix) }}`